### ISSUE TITLE
Could you share the resulting picture?

### ISSUE THREAD

**Issue by SkyfengBiuBiu** [varunagrawal/tiny-faces-pytorch#4]
Hello, 
At first, thank you for your implement.
There are still some bugs which I hope you could fix for better project.

1. In the "WIDERFace" class, the load function would cause an error while reading the the pictures without bounding boxes. It would be fixed if it is changed to :

                bboxes = np.empty((n, 10))
                
                if n!=0:
                    for b in range(n):
                        bboxes[b, :] = [abs(float(x))
                                    for x in lines[idx].strip().split()]
                        idx += 1
                else:
                    idx += 1

2. Then the PyTorch version in "requirement.txt" should be 0.4.1. 
3. The run function in "evaluate.py" has the debug option. In this place, you forgot unnormalize the image. Changing to :
        if debug:
            mean = torch.as_tensor([0.485, 0.456, 0.406], dtype=torch.float32, device=img.device)
            std = torch.as_tensor([0.229, 0.224, 0.225], dtype=torch.float32, device=img.device)
            
            img=(img[0]).permute((1, 2, 0))
            img=std*img+mean
            img=np.clip(img.numpy(),0,1)
            im = Image.fromarray((img*255).astype('uint8'), 'RGB')
            visualize.visualize_bboxes(im, dets)

it would fix the wired picture.

What's more? After applying the "visualize.visualize_bboxes" , the resulting picture (from test folder) does not satisfy me as the original Matlab model. By using the default parameters as you gave and training model with 50 epochs, the results are presented as the following figures.
![0 49843369913251623](https://user-images.githubusercontent.com/28752150/61381962-d932b900-a879-11e9-8993-1f7fabce9eca.jpg)
![0 26791034629033217](https://user-images.githubusercontent.com/28752150/61381963-d932b900-a879-11e9-8ea5-44bbe0d1b8e2.jpg)
![0 12144445914391433](https://user-images.githubusercontent.com/28752150/61381964-d9cb4f80-a879-11e9-95f5-04d96474d0ae.jpg)
![0 9841258305625651](https://user-images.githubusercontent.com/28752150/61381966-d9cb4f80-a879-11e9-982c-d7c2fcecb5c9.jpg)
![0 8950883280463936](https://user-images.githubusercontent.com/28752150/61381969-d9cb4f80-a879-11e9-9447-89cc56f9b5f7.jpg)
![0 8476592584406182](https://user-images.githubusercontent.com/28752150/61381970-d9cb4f80-a879-11e9-9b04-2b63eca40fd3.jpg)
![0 6770404203657887](https://user-images.githubusercontent.com/28752150/61381971-d9cb4f80-a879-11e9-91b6-265b463785a2.jpg)
![0 5918913651406368](https://user-images.githubusercontent.com/28752150/61381972-da63e600-a879-11e9-8cef-98e382cfd600.jpg)
![0 3623791789569477](https://user-images.githubusercontent.com/28752150/61381973-da63e600-a879-11e9-8a3d-a7f4a2c16d63.jpg)

(prob_thresh=0.6 nms_thresh=0.1)
Could you provide your per-trained model and share us the resulting pictures?
Thank you very much!!
For better project :+1: :+1:

**Comment by varunagrawal** [comment]
@SkyfengBiuBiu thank you for the feedback.

1. If you have `n=0` in the dataset, the next line should be blank, and not have any values, else the dataset is incorrect. I tested the use case you defined and the code works fine, especially since the loop will not execute if n is 0.

2. I have updated the requirements.txt. Thank you for that.

3. That debug code was actually for my convenience to debug some weird behavior. I have removed it in the latest commits and updated the code. If you run the current master branch, you should get far better results than what you have been getting. There was a bug where I was normalizing the image twice, but I have fixed that and things work great (I am able to get close to the same results as the original implementation from the paper author).

**Comment by Queequeg92** [comment]
@varunagrawal 
Hi, thanks for your good work!
If you have no bounding box in an image, the next line is blank or all zeros in wider face ground-truth files.
You have to delete all zeros cases manually, or you will get an error.

**Comment by varunagrawal** [comment]
I haven't done so in my dataloaders and both train and validation run fine. I am not sure why you would get an error. Can you please provide more details? Things like error traces, line numbers in the data files etc?

**Comment by varunagrawal** [comment]
Any updates?

**Comment by hamzakhalil798** [comment]
@SkyfengBiuBiu  can you kindly share the visualization script for an external picture or a picture from the test dataset?

And Did you find or trained a better model if yes please share
